### PR TITLE
Longest substring

### LINKED ISSUE #160: Longest Substring
### CP
<hr>

Given a string `s` consisting of lowercase letters, 
find the length of the longest substring that contains at most `k` distinct characters.

**Input Format**

`s`  a string where 1 <= len(s) <= 10<sup>4</sup>
`k`  an integer where 1 <= k <= 26

**Output Format**

Print an integer representing the length of the longest substring that contains at most k distinct characters.

Sample Test Case

**Input**
aabbbc
2
**Output**
5
**Explaination**

<i>In the above test case the longest substring containing at most 2 distinct characters is </i>`aabbb` <i> , containing only 2 distinct characters </i> `a` and `b` , <i>having a total length of 5</i>

### PR DESCRIPTION
**Pull request by ragss69**

### CODE REVIEW

**Comment by ragss69** [comment]
Please review my PR so I can be assigned more issues...

**Comment by Ishwarendra** [comment]
@ragss69 mention issue number in commit.

**Comment by ragss69** [comment]
> @ragss69 mention issue number in commit.

Done

File: cp/longest substring.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+s=str(input())
+k=int(input())
+d={}
+length=0
+if (1 <= len(s) <= 10**4) and (1 <= k <= 26):
```
Comment (ken1000minus7) [inline cp/longest substring.py]: You don't need this if condition, the constraints mean that u can assume that user's input iwll be in this range, so there's no need to check if the input lies in this range

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]
Your logic is incorrect
```
ababab
1
```
For this testcase answer is 1, but your code gives 3

**Comment by ragss69** [comment]
> Your logic is incorrect
> 
> ```
> ababab
> 1
> ```
> 
> For this testcase answer is 1, but your code gives 3

Sir  we have to find the length of the longest substring and according to the test case,  substring of 'ababab',  having one distinct character will be 'aaa' or 'bbb'and its length is 3.....please correct me if I am wrong or misunderstood the question.

**Comment by ken1000minus7** [comment]
by definition, a substring is a **contiguous** sequence of characters within a string, so in `ababab` in `ab` is a substring as it is the string formed from the sequence of characters from index 0 to 1 while `aaa` is not a substring as no 3 a's occur together in the string

**Comment by ragss69** [comment]
> by definition, a substring is a **contiguous** sequence of characters within a string, so in `ababab` in `ab` is a substring as it is the string formed from the sequence of characters from index 0 to 1 while `aaa` is not a substring as no 3 a's occur together in the string

Understood sir, I have now made the required changes...... please check

**Comment by ken1000minus7** [comment]
```
aba
3
```
Answer would be 3, but ur code gives runtime error, the questions says that no of unique characters should be at most k, not exactly k

**Comment by ragss69** [comment]
> ```
> aba
> 3
> ```
> 
> Answer would be 3, but ur code gives runtime error, the questions says that no of unique characters should be at most k, not exactly k

Changed "==k "to "<=k", hope that would fix the problem...

**Review by ken1000minus7 — APPROVED** [review]
Lgtm

**Pull request merged**